src/tests/ftest/performance/performance_test_base.py

```diff
@@ -0,0 +1,484 @@
+
+        """
+        if stop_delay is not None and (stop_delay < 0 or stop_delay > 1):
+            self.fail("stop_delay must be between 0 and 1")
+
+        if namespace is not None:
+            self.mdtest_cmd.namespace = namespace
+            self.mdtest_cmd.get_params(self)
+            self.set_processes_ppn(namespace)
+
+        # Performance with POSIX/DFUSE is tricky because we can't just set
+        # dfs_dir_oclass and dfs_oclass. This needs more work before running non-DFS.
+        if self.mdtest_cmd.api.value != 'DFS':
+            self.fail("Only DFS API supported")
+
+        stop_after = (stop_delay or 0) * (self.mdtest_cmd.stonewall_timer.value or 0)
+
+        self._log_performance_params("mdtest")
+
+        self.verify_oclass_server_count(self.mdtest_cmd.dfs_oclass.value)
+        self.verify_oclass_server_count(self.mdtest_cmd.dfs_dir_oclass.value)
+
+        # Set the container redundancy factor to match the oclass used
+        cont_rf = min([
+            oclass_utils.extract_redundancy_factor(self.mdtest_cmd.dfs_oclass.value),
+            oclass_utils.extract_redundancy_factor(self.mdtest_cmd.dfs_dir_oclass.value)])
+
+        # Create pool and container upfront so rank stop timing is more accurate
+        self.add_pool(connect=False)
+        self.add_container(self.pool, create=False)
+        self.container.chunk_size.update(self.mdtest_cmd.dfs_chunk.value)
+        properties = "rf:{}".format(cont_rf)
+        current_properties = self.container.properties.value
+        if current_properties:
+            new_properties = current_properties + "," + properties
+        else:
+            new_properties = properties
+        self.container.properties.update(new_properties)
+        self.container.create()
+
+        # Never let execute_mdtest automatically destroy the container
+        self.mdtest_cmd.dfs_destroy.update(False)
+
+        # Always run as a subprocess so we can stop ranks during IO
+        self.subprocess = True
+
+        self.log.info("Running MDTEST")
+        self.execute_mdtest()
+        if stop_after:
+            time.sleep(stop_after)
+            self.server_managers[0].stop_random_rank(self.d_log, force=True, exclude_ranks=[0])
+        mdtest_returncode = self.job_manager.process.wait()
+        try:
+            if mdtest_returncode != 0:
+                self.fail("mdtest failed")
+            mdtest_output = get_subprocess_stdout(self.job_manager.process)
+            mdtest_metrics = MdtestMetrics(mdtest_output)
+            if not mdtest_metrics:
+                self.fail("Failed to get mdtest metrics")
+            self.log_performance([
+                "create_ops: {}".format(mdtest_metrics.rates.file_creation.max),
+                "stat_ops: {}".format(mdtest_metrics.rates.file_stat.max),
+                "read_ops: {}".format(mdtest_metrics.rates.file_read.max),
+                "remove_ops: {}".format(mdtest_metrics.rates.file_removal.max)
```

Comment:
(pylint-no-member) Instance of 'MdtestMetricsGroup' has no 'file_removal' member